src/main/java/nextstep/subway/line/domain/Section.java

```diff
@@ -1,45 +1,88 @@
 package nextstep.subway.line.domain;
 
+import nextstep.subway.common.BaseEntity;
+import nextstep.subway.line.exception.SectionExceptionCode;
 import nextstep.subway.station.domain.Station;
 
 import javax.persistence.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
 
 @Entity
-public class Section {
+public class Section extends BaseEntity {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
     @ManyToOne(cascade = CascadeType.PERSIST)
-    @JoinColumn(name = "line_id")
+    @JoinColumn(nullable = false)
     private Line line;
 
     @ManyToOne(cascade = CascadeType.PERSIST)
-    @JoinColumn(name = "up_station_id")
+    @JoinColumn(nullable = false)
     private Station upStation;
 
     @ManyToOne(cascade = CascadeType.PERSIST)
-    @JoinColumn(name = "down_station_id")
+    @JoinColumn(nullable = false)
     private Station downStation;
 
-    private int distance;
+    @Embedded
+    private Distance distance;
 
-    public Section() {
+    protected Section() {
     }
 
     public Section(Line line, Station upStation, Station downStation, int distance) {
+        validate(line, upStation, downStation);
+
         this.line = line;
         this.upStation = upStation;
         this.downStation = downStation;
-        this.distance = distance;
+        this.distance = new Distance(distance);
+    }
+
+    private void validate(Line line, Station upStation, Station downStation) {
+        validateLine(line);
+        validateStations(upStation, downStation);
+    }
+
+    private void validateLine(Line line) {
+        if(Objects.isNull(line)) {
+            throw new IllegalArgumentException(SectionExceptionCode.REQUIRED_LINE.getMessage());
+        }
+    }
+
+    private void validateStations(Station upStation, Station downStation) {
+        if(Objects.isNull(upStation)) {
+            throw new IllegalArgumentException(SectionExceptionCode.REQUIRED_UP_STATION.getMessage());
+        }
+
+        if(Objects.isNull(downStation)) {
+            throw new IllegalArgumentException(SectionExceptionCode.REQUIRED_DOWN_STATION.getMessage());
+        }
+
+        if(upStation.equals(downStation)) {
+            throw new IllegalArgumentException(
+                    SectionExceptionCode.CANNOT_BE_THE_SAME_EACH_STATION.getMessage());
+        }
+    }
+
+    void updateLine(Line line) {
+        if(this.line != line) {
+            this.line = line;
+            line.addSection(this);
+        }
     }
 
-    public Long getId() {
-        return id;
+    public void divideSection(Section request) {
+        this.distance = distance.minus(new Distance(request.getDistance()));
```

Comment:
```suggestion
        this.distance = distance.minus(request.getDistance());
```
`getDistance`가 `distance`를 그대로 리턴하면 이렇게 쓸수 있을것 같네요.

또는 같은 `Section` 클래스의 값이니
```suggestion
        this.distance = distance.minus(request.distance);
```
이렇게 할수도 있겠네요! 😀

Reply:
깔끔하게 개선해주셔서 감사합니다! 많이 배워가네요 ㅎㅎ